src/wp-includes/user.php

```diff
@@ -3012,19 +3012,57 @@ function retrieve_password( $user_login = null ) {
 	 */
 	$message = apply_filters( 'retrieve_password_message', $message, $key, $user_login, $user_data );
 
+	// Wrap the single notification email arguments in an array to pass them to the retrieve_password_notification_email filter.
+	$retrieve_password_notification_email = array(
+		'to'      => $user_email,
+		'subject' => $title,
+		'message' => $message,
+		'headers' => '',
+	);
+
+	/**
+	 * Filters the contents of the reset password notification email sent to the user.
+	 *
+	 * @since x.x.x
+	 *
+	 * @param array   $retrieve_password_notification_email {
+	 *     Used to build wp_mail().
+	 *
+	 *     @type string $to      The intended recipient - user email address.
+	 *     @type string $subject The subject of the email.
+	 *     @type string $message The body of the email.
+	 *     @type string $headers The headers of the email.
+	 * }
+	 * @param string  $key        The activation key.
+	 * @param string  $user_login The username for the user.
+	 * @param WP_User $user_data  WP_User object.
+	 */
+	$retrieve_password_notification_email = apply_filters( 'retrieve_password_notification_email', $retrieve_password_notification_email, $key, $user_login, $user_data );
```

Comment:
This line is pretty long.

Some things we could do:

- Put `$data = compact( 'key', 'user_login', 'user_data' );` just above the docblock. This makes adding data in future easier too.
- Rename the initial `$retrieve_password_notification_email` to `$defaults`, immediately indicating they are not final.
- Rename the filtered `$retrieve_password_notification_email` to `$notification_email`. The context is implied.

This would result in a reduction of 70 characters in line length, and make adding more data to the filter easier.

```php
	// Current
	$retrieve_password_notification_email = apply_filters( 'retrieve_password_notification_email', $retrieve_password_notification_email, $key, $user_login, $user_data );

	// Reduced by 70 characters.
	$notification_email = apply_filters( 'retrieve_password_notification_email', $defaults, $data );
```

Reply:
Sounds great to me!

Regarding point one: If we put the parameters in an array the filter would be not consistent with  wp_password_change_notification and wp_new_user_notification. But I also think we should go for it, because of the reason you mentioned.

Reply:
It's true that it would make this filter inconsistent with `wp_password_change_notification` and `wp_new_user_notification`.

However, we use `compact` in numerous filters in Core and even in [wp_mail()](https://core.trac.wordpress.org/browser/tags/5.8.1/src/wp-includes/pluggable.php#L200) itself, so I think that validates our reasoning for using `compact` here to keep things tidy.

Hmm... I think it's even worth having an additional ticket at some point to bring the above mentioned functions in line with this for consistency and readability. No one likes a horizontal scrollbar, and not everyone has ultra-wide monitors.

Reply:
Okay, I implemented `compact`.

Reply:
I agree that `compact` should be implemented in the mentioned filters too.